riscv-syscalls/src/lib.rs

```diff
@@ -55,4 +55,6 @@ syscalls!(
     (7, KeccakF, "keccakf"),
     (8, Mod256, "mod_256"),
     (9, Halt, "halt"),
+    (10, CommitPublic, "commit_public"),
+    (11, PoseidonNative, "poseidon_native"),
```

Comment:
Has to be poseidon? Can't be some other permutation function?

For the lack of a better name, I suggest "NativeCryptoPermutation".

Reply:
currently it will be poseidon, and only poseidon for now has that property, so why not name it accurately?

Reply:
So, poseidon2 is a kind of poseidon?

Reply:
Also, Keccak fits the bill, and might be the fastest on some field (binary field?)

Reply:
I'll go with NativeHash for now, I think it's a bit nicer